### DISCUSSION TITLE
Como se tornar um desenvolvedor melhor?

### DISCUSSION THREAD

**Discussion by GuilhermeLis** [frontendbr/forum#1904] [Discussão]
Uma de minhas maiores dificuldade é como me tornar um desenvolvedor melhor, como eu posso melhorar minhas habilidades para conseguir me tornar um desenvolvedor sênior um dia. Além disso umas das coisas que me deixam muito confuso seria **como eu faço para as empresas me notarem eu preciso ter um currículo bacana? deveria fazer um site portfólio? ter muitos projetos feitos?**

Hoje eu trabalho como PJ para uma empresa e estou gostando muito, mas não quero ficar dependente não quero ficar afundado nisso.

**Como vocês sabem das novidades da área?** 

Outra coisa, eu gosto muito de ser fullstack, mas quando eu terminar a faculdade eu quero começar a trabalhar com data Science, com deep learn, algo desse jeito.

Esse é meu primeiro emprego como desenvolvedor.

**Comment by wmartins** [comment]
Acho que é bem difícil a resposta para essas perguntas, então proponho que a gente vá construíndo isso aos poucos, com algumas ideias e alguns pontos.

Casualmente hoje eu li um texto muito bom, chamado [_How to Build a Strong Career in Tech_](https://www.linkedin.com/pulse/how-build-strong-career-tech-thiago-ghisi/). É uma leitura longa, mas vale cada pedaço, acho que pode ser um material legal.

Agora vou tentar pegar alguns pontos que tu trouxe:

> como eu posso melhorar minhas habilidades para conseguir me tornar um desenvolvedor sênior um dia

Acho que o ponto de partida aqui é: o que é ser sênior?

Aqui no próprio forum tu vai conseguir achar boas discussões:

- https://github.com/frontendbr/forum/issues/92
- https://github.com/frontendbr/forum/issues/252

Eu acho particularmente muito difícil de _carimbar_ essas nomenclaturas para as pessoas. Uma das minhas funções hoje é ajudar meu time de recrutamento com os processos seletivos, e o pessoal sempre me pergunta qual é o nível de senioridade e eu tento ao máximo não dizer o nível, mas dizer coisas do tipo: _essa pessoa vai precisar de ajuda com o desenvolvimento técnico_; _acho que essa pessoa consegue se virar sozinha, mas precisa aprender mais coisas_; _essa pessoa tem capacidade para desenvolver outras_.

Se posso te dar um conselho: não se apega tanto com esses termos, o ideal é identificar quais são os pontos que tu precisa trabalhar, e é importante que esses pontos não sejam somente técnicos. Acho que boa parte do trabalho de pessoas que sinto que estão em posições de maior senioridade são trabalhos que envolvem comunicação e alinhamentos. Ao menos nas minhas experiências, sempre admirei o pessoal que conseguia _navegar_ em vários contextos para explorar algumas soluções.

> como eu faço para as empresas me notarem eu preciso ter um currículo bacana? deveria fazer um site portfólio? ter muitos projetos feitos?

Tu ter um currículo legal, site, portfolio, projetos e etc são coisas que somam, com certeza, mas ao meu ver não são fundamentais. Nos últimos anos eu venho entrevistando muita gente. Já vi gente muito boa com bem pouquinha coisa no currículo e quase nada no GitHub e já vi gente que não tinha o que a vaga precisava, embora tivesse um currículo muito bem feito e vários projetos pra mostrar.

Acho que um outro ponto importante é que as vezes a gente foca muito no conhecimento técnico e esquece que também precisamos demonstrar _conhecimentos pessoais_. Eu sempre digo para os meus colegas que eu quero contratar _gente boa_, mas não só no sentido técnico, no sentido de ser uma pessoa boa de se trabalhar mesmo. Aquela pessoa que quando alguém precisa de ajuda, vai ser a primeira a levantar a mão para ajudar. Vi um colega uma vez levar uma garrafa de café recém feito para algumas pessoas que estavam quebrando a cabeça com um problema em produção. Isso pra mim é um exemplo de como ser um bom colega de equipe, mesmo não estando diretamente envolvido em algo. E acho que nessa linha tem outras coisas, admitir não saber algo, sem tentar enrolar, se mostrar disposto à ir atrás e por aí vai.

Claro que as habilidades técnicas são importantes, e são elas que vão te ajudar a resolver os problemas, porém, acho que é importante um balanço, conseguir entender o contexto das pessoas, o contexto da empresa/projeto que tu tá trabalhando e por aí vai também são importantes. Pegando algumas frases do texto que linkei ali em cima, acho que isso aqui dá uma boa ideia desse ponto:

> _But, the not so obvious advice that I have learned and has influenced my career in the last couple of years is to Maximize the Work Not Done. Maximize the number of lines not written._

> _Don't focus on being efficient, focus on being effective. You must learn how to prioritize, that is the only way to be highly effective. Efficiency is the tweaks that you make to get that 10% optimization. 90% of productivity is about being effective._

Por fim,

> Como vocês sabem das novidades da área?

Normalmente eu sigo algumas pessoas no Twitter, tento acompanhar algo no Reddit e assino algumas listas de e-mail.

---

Espero ter contribuído, caso queira trocar mais alguma ideia sobre algum ponto, fico à disposição! Boa jornada!
